chain/bitcoind_conn.go

```diff
@@ -139,10 +140,32 @@ func (c *BitcoindConn) Start() error {
 		return nil
 	}
 
+	var (
+		net wire.BitcoinNet
+		err error
+	)
+
 	// Verify that the node is running on the expected network.
-	net, err := c.getCurrentNet()
-	if err != nil {
-		return err
+	btcInWarmup := false
+	for {
+		net, err = c.getCurrentNet()
+		if err == nil {
+			// No error means bitcoind is responsive and we can proceed.
+			break
+		} else if err.(*btcjson.RPCError).Code == btcjson.ErrRPCInWarmup {
```

Comment:
We should do a soft type assertion here to prevent a panic if `err` is not of the type `*btcjson.RPCError`:
```
} else if rpcErr, ok := err.(*btcjson.RPCError); ok && rpcErr.Code == btcjson.ErrRPCInWarmup {
```

Reply:
Thanks, makes sense. Done.